### PR TITLE
more flexible default sanitizer pattern

### PR DESCRIPTION
**Pull request by jeffkala**
# Closes:  NA
# What's Changed

Default sanitizer patterns don't capture the same patterns wrapped in a python dictionary. Also secret / secrets isn't a pattern that is currently caught.  This is a small update to just be a slighly more flexible default sanitizer.

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Comment by timizuoebideri1** [comment]
Could you add a test case entry to `nautobot.core.tests.test_logging.LoggingUtilitiesTest.DIRTY_CLEAN`

File: nautobot/core/settings.py [hunk 1]
```diff
@@ -124,7 +124,10 @@
 SANITIZER_PATTERNS = [
     # General removal of username-like and password-like tokens
     (re.compile(r"(https?://)?\S+\s*@", re.IGNORECASE), r"\1{replacement}@"),
-    (re.compile(r"(username|password|passwd|pwd)((?:\s+is.?|:)?\s+)\S+", re.IGNORECASE), r"\1\2{replacement}"),
+    (
+        re.compile(r"(username|password|passwd|pwd|secret|secrets)([\"']?(?:\s+is.?|:)?\s+)\S+[\"']?", re.IGNORECASE),
```
Comment (gsnider2195) [inline nautobot/core/settings.py:128]: Probably also a good idea to provide an example of what kind of cases this covers in a comment

Comment (jeffkala) [reply nautobot/core/settings.py:128]: @gsnider2195  In a comment in the PR or in the doc string?

Comment (glennmatthews) [reply nautobot/core/settings.py:128]: Personally I'd vote for a comment here and maybe a note in the documentation as well.

**Comment by jeffkala** [comment]
Opened an issue #5031 for proper changelog entry.

File: nautobot/core/settings.py (hunk 1 above)
Comment (glennmatthews) [reply nautobot/core/settings.py:128]: @jeffkala The docs update is the only thing remaining before we can merge this, I think?

Comment (jeffkala) [reply nautobot/core/settings.py:128]: will get the docs added today.

Comment (jeffkala) [reply nautobot/core/settings.py:128]: @glennmatthews fit best to put some additional docs in the optional-settings existing docs? Or create a new page of sorts?  Not seeing anywhere else that fits in the current docs structure.

Comment (glennmatthews) [reply nautobot/core/settings.py:128]: Yeah I was thinking in the optional-settings docs for SANITIZER_PATTERNS.

**Review by glennmatthews — APPROVED** [review]

File: nautobot/core/settings.py (hunk 1 above)
Comment (jeffkala) [reply nautobot/core/settings.py:128]: Just updated optional settings with some examples.

**Pull request merged**